### DISCUSSION TITLE
Deadkeys locks cursor in place

### DISCUSSION THREAD

**Discussion by filipeaguiar** [ghostty-org/ghostty#3164] [Feature Requests, Ideas] [answered]
Any deadkey that I press locks the cursor in place. No matter what I press after that the character under cursor changes but the terminal becomes unusable.
 ![Image](https://github.com/user-attachments/assets/9f2d5b13-23b7-4c7d-8be0-6c6357f2efea)

**Comment by mitchellh** [comment]
Moved to a discussion since this isn't immediately actionable yet (see CONTRIBUTING.md). I feel like we'll get there quickly though. Some questions:

1. `ghostty +version` output 
2. What keyboard layout?
3. Can you document keys to input to reproduce this?

Thank you!

**Reply by llynch** [reply]
I think this is the same bug as this issue:
https://github.com/ghostty-org/ghostty/issues/3567

**Comment by filipeaguiar** [comment]
`ghostty +version`

```
Ghostty 1.0.0-main+4b4d4062

Version
  - version: 1.0.0-main+4b4d4062
  - channel: tip
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.gtk
  - font engine: font.main.Backend.fontconfig_freetype
  - renderer   : renderer.OpenGL
  - libxev     : main.Backend.io_uring
  - GTK version:
    build      : 4.14.2
    runtime    : 4.14.2
  - libadwaita : enabled
    build      : 1.5.0
    runtime    : 1.5.0
```

Keyboard Layout: English (with dead keys)

The issue happens every time I press a deadkey: ' " ^ ` ~ 

That's my config file if it helps:
font-family = 
font-family-bold = 
font-family-italic = 
font-family-bold-italic = 
font-style = default
font-style-bold = default
font-style-italic = default
font-style-bold-italic = default
font-synthetic-style = bold,italic,bold-italic
font-feature = 
font-size = 12
font-variation = 
font-variation-bold = 
font-variation-italic = 
font-variation-bold-italic = 
font-codepoint-map = 
font-thicken = false
adjust-cell-width = 
adjust-cell-height = 
adjust-font-baseline = 
adjust-underline-position = 
adjust-underline-thickness = 
adjust-strikethrough-position = 
adjust-strikethrough-thickness = 
adjust-overline-position = 
adjust-overline-thickness = 
adjust-cursor-thickness = 
adjust-cursor-height = 
adjust-box-thickness = 
grapheme-width-method = unicode
freetype-load-flags = hinting,force-autohint,monochrome,autohint
theme = catppuccin-mocha
selection-foreground = 
selection-background = 
selection-invert-fg-bg = false
minimum-contrast = 1
palette = 0=#1d1f21
palette = 1=#cc6666
palette = 2=#b5bd68
palette = 3=#f0c674
palette = 4=#81a2be
palette = 5=#b294bb
palette = 6=#8abeb7
palette = 7=#c5c8c6
palette = 8=#666666
palette = 9=#d54e53
palette = 10=#b9ca4a
palette = 11=#e7c547
palette = 12=#7aa6da
palette = 13=#c397d8
palette = 14=#70c0b1
palette = 15=#eaeaea
palette = 16=#000000
palette = 17=#00005f
palette = 18=#000087
palette = 19=#0000af
palette = 20=#0000d7
palette = 21=#0000ff
palette = 22=#005f00
palette = 23=#005f5f
palette = 24=#005f87
palette = 25=#005faf
palette = 26=#005fd7
palette = 27=#005fff
palette = 28=#008700
palette = 29=#00875f
palette = 30=#008787
palette = 31=#0087af
palette = 32=#0087d7
palette = 33=#0087ff
palette = 34=#00af00
palette = 35=#00af5f
palette = 36=#00af87
palette = 37=#00afaf
palette = 38=#00afd7
palette = 39=#00afff
palette = 40=#00d700
palette = 41=#00d75f
palette = 42=#00d787
palette = 43=#00d7af
palette = 44=#00d7d7
palette = 45=#00d7ff
palette = 46=#00ff00
palette = 47=#00ff5f
palette = 48=#00ff87
palette = 49=#00ffaf
palette = 50=#00ffd7
palette = 51=#00ffff
palette = 52=#5f0000
palette = 53=#5f005f
palette = 54=#5f0087
palette = 55=#5f00af
palette = 56=#5f00d7
palette = 57=#5f00ff
palette = 58=#5f5f00
palette = 59=#5f5f5f
palette = 60=#5f5f87
palette = 61=#5f5faf
palette = 62=#5f5fd7
palette = 63=#5f5fff
palette = 64=#5f8700
palette = 65=#5f875f
palette = 66=#5f8787
palette = 67=#5f87af
palette = 68=#5f87d7
palette = 69=#5f87ff
palette = 70=#5faf00
palette = 71=#5faf5f
palette = 72=#5faf87
palette = 73=#5fafaf
palette = 74=#5fafd7
palette = 75=#5fafff
palette = 76=#5fd700
palette = 77=#5fd75f
palette = 78=#5fd787
palette = 79=#5fd7af
palette = 80=#5fd7d7
palette = 81=#5fd7ff
palette = 82=#5fff00
palette = 83=#5fff5f
palette = 84=#5fff87
palette = 85=#5fffaf
palette = 86=#5fffd7
palette = 87=#5fffff
palette = 88=#870000
palette = 89=#87005f
palette = 90=#870087
palette = 91=#8700af
palette = 92=#8700d7
palette = 93=#8700ff
palette = 94=#875f00
palette = 95=#875f5f
palette = 96=#875f87
palette = 97=#875faf
palette = 98=#875fd7
palette = 99=#875fff
palette = 100=#878700
palette = 101=#87875f
palette = 102=#878787
palette = 103=#8787af
palette = 104=#8787d7
palette = 105=#8787ff
palette = 106=#87af00
palette = 107=#87af5f
palette = 108=#87af87
palette = 109=#87afaf
palette = 110=#87afd7
palette = 111=#87afff
palette = 112=#87d700
palette = 113=#87d75f
palette = 114=#87d787
palette = 115=#87d7af
palette = 116=#87d7d7
palette = 117=#87d7ff
palette = 118=#87ff00
palette = 119=#87ff5f
palette = 120=#87ff87
palette = 121=#87ffaf
palette = 122=#87ffd7
palette = 123=#87ffff
palette = 124=#af0000
palette = 125=#af005f
palette = 126=#af0087
palette = 127=#af00af
palette = 128=#af00d7
palette = 129=#af00ff
palette = 130=#af5f00
palette = 131=#af5f5f
palette = 132=#af5f87
palette = 133=#af5faf
palette = 134=#af5fd7
palette = 135=#af5fff
palette = 136=#af8700
palette = 137=#af875f
palette = 138=#af8787
palette = 139=#af87af
palette = 140=#af87d7
palette = 141=#af87ff
palette = 142=#afaf00
palette = 143=#afaf5f
palette = 144=#afaf87
palette = 145=#afafaf
palette = 146=#afafd7
palette = 147=#afafff
palette = 148=#afd700
palette = 149=#afd75f
palette = 150=#afd787
palette = 151=#afd7af
palette = 152=#afd7d7
palette = 153=#afd7ff
palette = 154=#afff00
palette = 155=#afff5f
palette = 156=#afff87
palette = 157=#afffaf
palette = 158=#afffd7
palette = 159=#afffff
palette = 160=#d70000
palette = 161=#d7005f
palette = 162=#d70087
palette = 163=#d700af
palette = 164=#d700d7
palette = 165=#d700ff
palette = 166=#d75f00
palette = 167=#d75f5f
palette = 168=#d75f87
palette = 169=#d75faf
palette = 170=#d75fd7
palette = 171=#d75fff
palette = 172=#d78700
palette = 173=#d7875f
palette = 174=#d78787
palette = 175=#d787af
palette = 176=#d787d7
palette = 177=#d787ff
palette = 178=#d7af00
palette = 179=#d7af5f
palette = 180=#d7af87
palette = 181=#d7afaf
palette = 182=#d7afd7
palette = 183=#d7afff
palette = 184=#d7d700
palette = 185=#d7d75f
palette = 186=#d7d787
palette = 187=#d7d7af
palette = 188=#d7d7d7
palette = 189=#d7d7ff
palette = 190=#d7ff00
palette = 191=#d7ff5f
palette = 192=#d7ff87
palette = 193=#d7ffaf
palette = 194=#d7ffd7
palette = 195=#d7ffff
palette = 196=#ff0000
palette = 197=#ff005f
palette = 198=#ff0087
palette = 199=#ff00af
palette = 200=#ff00d7
palette = 201=#ff00ff
palette = 202=#ff5f00
palette = 203=#ff5f5f
palette = 204=#ff5f87
palette = 205=#ff5faf
palette = 206=#ff5fd7
palette = 207=#ff5fff
palette = 208=#ff8700
palette = 209=#ff875f
palette = 210=#ff8787
palette = 211=#ff87af
palette = 212=#ff87d7
palette = 213=#ff87ff
palette = 214=#ffaf00
palette = 215=#ffaf5f
palette = 216=#ffaf87
palette = 217=#ffafaf
palette = 218=#ffafd7
palette = 219=#ffafff
palette = 220=#ffd700
palette = 221=#ffd75f
palette = 222=#ffd787
palette = 223=#ffd7af
palette = 224=#ffd7d7
palette = 225=#ffd7ff
palette = 226=#ffff00
palette = 227=#ffff5f
palette = 228=#ffff87
palette = 229=#ffffaf
palette = 230=#ffffd7
palette = 231=#ffffff
palette = 232=#080808
palette = 233=#121212
palette = 234=#1c1c1c
palette = 235=#262626
palette = 236=#303030
palette = 237=#3a3a3a
palette = 238=#444444
palette = 239=#4e4e4e
palette = 240=#585858
palette = 241=#626262
palette = 242=#6c6c6c
palette = 243=#767676
palette = 244=#808080
palette = 245=#8a8a8a
palette = 246=#949494
palette = 247=#9e9e9e
palette = 248=#a8a8a8
palette = 249=#b2b2b2
palette = 250=#bcbcbc
palette = 251=#c6c6c6
palette = 252=#d0d0d0
palette = 253=#dadada
palette = 254=#e4e4e4
palette = 255=#eeeeee
cursor-color = 
cursor-invert-fg-bg = false
cursor-opacity = 1
cursor-style = block
cursor-style-blink = 
cursor-text = 
cursor-click-to-move = true
mouse-hide-while-typing = false
mouse-shift-capture = false
mouse-scroll-multiplier = 1
background-opacity = 1
background-blur-radius = 0
unfocused-split-opacity = 0.7
unfocused-split-fill = 
command = 
initial-command = 
wait-after-command = false
abnormal-command-exit-runtime = 250
scrollback-limit = 10000000
link-url = true
fullscreen = true
title = 
class = 
x11-instance-name = 
working-directory = 
keybind = ctrl+comma=open_config
keybind = ctrl+alt+up=goto_split:top
keybind = ctrl+page_down=next_tab
keybind = ctrl+shift+v=paste_from_clipboard
keybind = shift+insert=paste_from_selection
keybind = ctrl+shift+a=select_all
keybind = shift+up=adjust_selection:up
keybind = alt+five=goto_tab:5
keybind = super+ctrl+right_bracket=goto_split:next
keybind = ctrl+equal=increase_font_size:1
keybind = ctrl+shift+o=new_split:right
keybind = ctrl+shift+c=copy_to_clipboard
keybind = ctrl+shift+q=quit
keybind = ctrl+shift+n=new_window
keybind = ctrl+shift+page_down=jump_to_prompt:1
keybind = ctrl+shift+comma=reload_config
keybind = ctrl+minus=decrease_font_size:1
keybind = shift+left=adjust_selection:left
keybind = super+ctrl+shift+up=resize_split:up,10
keybind = alt+eight=goto_tab:8
keybind = shift+page_up=scroll_page_up
keybind = ctrl+alt+shift+j=write_scrollback_file:open
keybind = ctrl+shift+left=previous_tab
keybind = ctrl+shift+w=close_surface
keybind = super+ctrl+shift+equal=equalize_splits
keybind = shift+end=scroll_to_bottom
keybind = ctrl+zero=reset_font_size
keybind = alt+three=goto_tab:3
keybind = ctrl+shift+j=write_scrollback_file:paste
keybind = ctrl+enter=toggle_fullscreen
keybind = ctrl+page_up=previous_tab
keybind = shift+right=adjust_selection:right
keybind = ctrl+tab=next_tab
keybind = ctrl+alt+left=goto_split:left
keybind = shift+page_down=scroll_page_down
keybind = ctrl+shift+right=next_tab
keybind = ctrl+shift+page_up=jump_to_prompt:-1
keybind = alt+nine=goto_tab:9
keybind = ctrl+shift+t=new_tab
keybind = shift+down=adjust_selection:down
keybind = super+ctrl+shift+left=resize_split:left,10
keybind = ctrl+shift+tab=previous_tab
keybind = alt+two=goto_tab:2
keybind = ctrl+alt+down=goto_split:bottom
keybind = super+ctrl+shift+down=resize_split:down,10
keybind = super+ctrl+shift+right=resize_split:right,10
keybind = ctrl+plus=increase_font_size:1
keybind = alt+four=goto_tab:4
keybind = ctrl+shift+e=new_split:down
keybind = ctrl+alt+right=goto_split:right
keybind = alt+f4=close_window
keybind = alt+one=goto_tab:1
keybind = ctrl+shift+enter=toggle_split_zoom
keybind = shift+home=scroll_to_top
keybind = super+ctrl+left_bracket=goto_split:previous
keybind = ctrl+shift+i=inspector:toggle
keybind = alt+six=goto_tab:6
keybind = alt+seven=goto_tab:7
window-padding-x = 2
window-padding-y = 2
window-padding-balance = false
window-padding-color = background
window-vsync = true
window-inherit-working-directory = true
window-inherit-font-size = true
window-decoration = false
window-title-font-family = 
window-theme = auto
window-colorspace = srgb
window-height = 0
window-width = 0
window-save-state = default
window-step-resize = false
window-new-tab-position = current
resize-overlay = after-first
resize-overlay-position = center
resize-overlay-duration = 750ms
focus-follows-mouse = false
clipboard-read = ask
clipboard-write = allow
clipboard-trim-trailing-spaces = true
clipboard-paste-protection = true
clipboard-paste-bracketed-safe = true
image-storage-limit = 320000000
copy-on-select = true
click-repeat-interval = 0
config-file = 
config-default-files = true
confirm-close-surface = true
quit-after-last-window-closed = true
quit-after-last-window-closed-delay = 
initial-window = true
quick-terminal-position = top
quick-terminal-screen = main
quick-terminal-animation-duration = 0.2
quick-terminal-autohide = true
shell-integration = detect
shell-integration-features = cursor,no-sudo,title
osc-color-report-format = 16-bit
vt-kam-allowed = false
custom-shader = 
custom-shader-animation = true
macos-non-native-fullscreen = false
macos-titlebar-style = transparent
macos-titlebar-proxy-icon = visible
macos-option-as-alt = 
macos-window-shadow = true
macos-auto-secure-input = true
macos-secure-input-indication = true
macos-icon = official
macos-icon-frame = aluminum
macos-icon-ghost-color = 
macos-icon-screen-color = 
linux-cgroup = single-instance
linux-cgroup-memory-limit = 
linux-cgroup-processes-limit = 
linux-cgroup-hard-fail = false
gtk-single-instance = desktop
gtk-titlebar = true
gtk-tabs-location = top
adw-toolbar-style = raised
gtk-wide-tabs = true
gtk-adwaita = true
desktop-notifications = true
bold-is-bright = false
term = xterm-ghostty
enquiry-response = 
auto-update = check
auto-update-channel =

**Comment by filipeaguiar** [comment]
Quick update: I've just tested the new comunity deb package in **ANOTHER** computer, with **ANOTHER** keyboard and layout. The problem happened again. Everytime I press a dead key, the cursor locks in place as shown in the gif before.

**Comment by mitchellh** [comment, marked as answer]
Lets move discussion to #3567
